.github/workflows/docker_build.yml

```diff
@@ -1,66 +1,66 @@
-      - name: Fetch Repo Name
-        id: repo-name
-        run: echo "::set-output name=value::$(echo 'gitpod4' | awk -F '/' '{print $2}')"
-      - name: Docker meta
+      # documentation at
+      # https://docs.github.com/en/actions/use-cases-and-examples/publishing-packages/publishing-docker-images
+      - uses: actions/checkout@v4
+      - name: Log in to the container registry
+        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3.4.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          # might need set as `secrets.DOCKER_USERNAME` and `secrets.DOCKER_PASSWORD`
+          # if using different registry than ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Extract metadata (tags, labels) for docker
         id: meta
         uses: docker/metadata-action@902fa8ec7d6ecbf8d84d538b9b233a880e428804 # v5.7.0
         with:
-          # list of Docker images to use as base name for tags
-          images: |
-            ghcr.io/leanprover-community/${{ github.event.repository.name }}
-          # generate Docker tags based on the following events/attributes
+          images: ${{ env.REGISTRY }}/${{ env.REPO_NAME }}/${{ matrix.image }}
           tags: |
+            type=schedule,pattern={{date 'YYYY-MM-DD'}}
+            type=sha
             type=ref,event=branch
             type=ref,event=tag
-            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
-      # cached builds according to
-      # https://github.com/docker/build-push-action/blob/d17cab8f420a2403aa182827208f8f957ccbf9a9/docs/advanced/cache.md#github-cache
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@b5ca514318bd6ebac0fb2aedd5d36ec1b5c232a2 # v3.10.0
-      - name: Login to GitHub Container Registry
-        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3.4.0
-        with:
-          # with future possibilities to do it to hub.docker.com
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Final tag name
-        id: output-tag-name
-        run: echo "::set-output name=tagname::${{ steps.meta.outputs.tags }}"
-      - name: Build and push image
+            type=raw,value=latest
+          labels: |
+            org.opencontainers.image.licenses=Apache-2.0
+      - name: Build and push docker image
         id: push
         uses: docker/build-push-action@471d1dc4e07e5cdedd4c2171150001c434f0b7a4 # v6.15.0
         with:
+          context: .
+          file: .docker/${{ matrix.image }}/Dockerfile
           push: true
-          # action can do checkout, but I don't know the exact subdir syntax
-          context: .docker/gitpod/
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
```

Comment:
I did not read up on caching now. Could be added again if desired.

Reply:
added the lines back blindly as they were. I'll assume they work?

Reply:
nope. @pechersky if you want to add caching back, feel free to push here. But can also be in a separate PR given that the docker images build quite fast without cache